src/lms-inspector.js

```diff
@@ -37,58 +39,130 @@ export const determineCompression = arrayBuffer => {
+      files[header.name] = '';
+      stream.on('data', data => { files[header.name] += data.toString(); });
+      stream.on('end', callback);
+    })
+    .on('finish', () => resolve(files));
 });
 
+/**
+ * Uses an imsmanifest.xml file to determine what LMS the archive is for
+ * @method checkManifest
+ * @memberof LMSInspector
+ * @param {String} manifest - the imsmanifest.xml file contents
+ * @returns {String} - the LMS type
+ */
+export const checkManifest = manifest => {
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(manifest, 'text/xml');
+  if (doc.getElementsByTagName('manifest')[0].getAttribute('identifier').includes('D2L')) {
+    return 'd2l';
+  }
+  return '';
+};
+
 /**
  * Looks at a list of files and determines which LMS they are for, if any
- * @method checkForLMS
+ * @method getType
  * @memberof LMSInspector
- * @param {Array.<String>} filenames - an array of filenames in an archive
- * @returns {Promise.<String>} - the LMS type
+ * @param {Object} files - an object with filenames and files
+ * @returns {String} - the LMS type
  */
-export const checkForLMS = filenames => new Promise((resolve, reject) => {
-  filenames.forEach(name => {
-    if (name.includes('course_settings')) resolve('canvas');
-    if (name.includes('moodle') || name.includes('completion.xml')) resolve('moodle');
-    if (name.includes('bb-')) resolve('blackboard');
-    if (name.includes('brainhoneymanifest')) resolve('buzz');
-    if (name.includes('d2l')) resolve('d2l');
-  });
+export const getType = files => {
+  const filenames = Object.keys(files);
+  for (let name of filenames) {
+    if (name.includes('course_settings/canvas_export.txt')) return 'canvas';
+    if (name.includes('moodle') || name.includes('completion.xml')) return 'moodle';
+    if (name.includes('brainhoneymanifest')) return 'buzz';
+    if (name.includes('imsmanifest.xml')) {
+      if (checkManifest(files[name]) === 'd2l') {
+        return 'd2l';
+      }
+      return 'blackboard';
+    }
+  }
 
-  reject('File is not an LMS archive');
-});
+  return Promise.reject('File is not an LMS archive');
+};
+
+/**
+ * Takes an LMS type, and the list of files in the archive, then looks through the files for information
+ * regarding the version
+ * @method getVersion
+ * @memberof LMSInspector
+ * @param {String} type - the LMS type
+ * @param {Object} files - an object containing the filenames and files in the archive
+ * @returns {String} - a string containing the version if found, or an empty string
+ */
+export const getVersion = (type, files) => {
+  switch (type) {
+    case 'moodle':
+      const parser = new DOMParser();
+      if (files['moodle.xml']) {
+        const doc = parser.parseFromString(files['moodle.xml'], 'text/xml');
+        return doc.getElementsByTagName('MOODLE_RELEASE')[0].childNodes[0].nodeValue;
+      }
+      if (files['moodle_backup.xml']) {
+        const doc = parser.parseFromString(files['moodle_backup.xml'], 'text/xml');
```

Comment:
Having the duplicate `const doc` declaration here may cause issues, I would recommend renaming them to something like `moodleDoc` and `moodleBackupDoc`.